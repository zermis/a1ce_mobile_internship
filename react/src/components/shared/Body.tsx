import React from 'react';
import { View, StyleSheet } from 'react-native';

import Colors from '../../constants/Colors';

interface Props {
    children?: JSX.Element;
    style?: object;
    accessibilityLabel?: string;
}

const Body = (props: Props): JSX.Element => {
    return (
        <View
            style={{ ...styles.container, ...props.style }}
            accessibilityLabel={props.accessibilityLabel}
        >
            {props.children}
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: Colors.bodyColor,
        borderTopLeftRadius: 30,
        borderTopRightRadius: 30,
        marginTop: -70
    }
});

export default Body;
