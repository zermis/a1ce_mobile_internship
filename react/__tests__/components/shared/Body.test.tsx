import { Text, View } from 'react-native';
import renderer from 'react-test-renderer';
import Body from '../../../src/components/shared/Body';

it('render body with items', () => {
    const tree = renderer
        .create(
            <Body>
                <View>
                    <Text>Hello World</Text>
                </View>
            </Body>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
