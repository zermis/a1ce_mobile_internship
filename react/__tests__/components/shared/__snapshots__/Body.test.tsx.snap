// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render body with items 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF8F8",
      "borderTopLeftRadius": 30,
      "borderTopRightRadius": 30,
      "flex": 1,
      "marginTop": -70,
    }
  }
>
  <View>
    <Text>
      Hello World
    </Text>
  </View>
</View>
`;
